pkg/reconciler/knativeserving/knativeserving.go

```diff
@@ -74,12 +72,14 @@ func (r *Reconciler) FinalizeKind(ctx context.Context, original *servingv1alpha1
 		}
 	}
 
-	// Appending nothing effectively results in a deep-copy clone
-	manifest := r.config.Append()
+	manifest, err := common.InstalledManifest(original)
+	if err != nil {
+		return err
+	}
+	manifest = r.manifest.Append(manifest)
```

Comment:
Reading this I'm starting to get a sense of the distinction between these two Manifest fetching methods.

So we initialize with the existing install, then apply the target install to it?

Reply:
The target manifest is what to install.
The installed manifest is the existing manifest, which has been installed. 
Since we about to support multiple versions of knative with one version of operator, installed manifest is what we have, and target manifest will be where we are going, in terms of upgrading and downgrading.

Reply:
Sounds reasonable, I still think it may be helpful to comment that here.

Reply:
Done